{
  try {
    analyze(session,query);
    fail(format("Expected error %s, but analysis succeeded",error));
  }
 catch (  SemanticException e) {
    if (e.getCode() != error) {
      fail(format("Expected error %s, but found %s: %s",error,e.getCode(),e.getMessage()),e);
    }
  }
}
